File: _designs_001/Two Sum III/Solution.java
public class Solution {
    public List<Boolean> testTwoSumIII(StreamTokenizer tokenizer, List<Double> metrics) throws IOException {
        TwoSum twoSum = new TwoSum();
        List<Boolean> result = new ArrayList<Boolean>();

        int opsCnt = Serializer.deserializeInt(tokenizer);
        for (int i = 0; i < opsCnt; i++) {
            int n;
            double startTime;

            switch (Serializer.deserializeInt(tokenizer)) {
            case 0:
                n = Serializer.deserializeInt(tokenizer);
                startTime = System.nanoTime();
                twoSum.add(n);
                metrics.set(0, metrics.get(0) + (double) (System.nanoTime() - startTime) / 1000000L);
                break;
            case 1:
                n = Serializer.deserializeInt(tokenizer);
                startTime = System.nanoTime();
                result.add(twoSum.find(n));
                metrics.set(0, metrics.get(0) + (double) (System.nanoTime() - startTime) / 1000000L);
                break;
            }
        }
        return result;
    }
}

File: _designs_001/Two Sum III/Solution.rb
class Solution
  def testTwoSumIII(inp, metrics)
    metrics[0] = 0
    two_sum, r = TwoSum.new, []
    Serializer.deserializeInt(inp).times do
      case Serializer.deserializeInt(inp)
      when 0
        n = Serializer.deserializeInt(inp)
        start_time = Time.now
        two_sum.add(n)
        metrics[0] += (Time.now - start_time) * 1000
      when 1
        n = Serializer.deserializeInt(inp)
        start_time = Time.now
        r << two_sum.find(n)
        metrics[0] += (Time.now - start_time) * 1000
      end
    end
    r
  end
end